### ISSUE TITLE
opendatahub-controller-manager-xxx pod deletion resets DSCInitialization namespace back to opendatahub

### ISSUE THREAD

**Issue by shalberd** [opendatahub-io/opendatahub-operator#724]
**Describe the bug**
operator version 2.4.0
Operator was created, DSCInitialization metadata.name "default" created with namespace "opendatahub" at

spec.applicationsNamespace
spec.monitoring.namespace

result: on first start of opendatahub-controller-manager-xxx pod in namespace openshift-operators,
the operator automatically creates the opendatahub namespace. 
Alright, so I delete that namespace, don't want it.

user change: I change namespace to another namespace by patching the default DSCInitialization with the new namespace name at spec.applicationsNamespace and spec.monitoring.namespace.

I create a DataScienceCluster, objects created in custom namespace, good.

**To Reproduce**
- delete the pod opendatahub-controller-manager-xxx in namespace openshift-operators
- observe pod coming up again, ok
- search for DSCInitialization, find default namespace "opendatahub" again in spec.applicationsNamespace
spec.monitoring.namespace   :-(

- find opendatahub namespace created again AND applications objects created in opendatahub namespace as well ...

**Expected behavior**
- main operator pod restarts happen frequently in openshift-operators (openshift updates, maintenance windows, VM node failure, you name it
- under no circumstances should the default DSCInitialization, once modified, be overwritten when the operator pod restarts

**Screenshots**

**Additional context**
With the way this works currently, not only do I have to delete an opendatahub namespace I do not want. In addition, the software deployed in a custom odh namespace and defined in DataScienceCluster is applied to the opendatahub namespace (which is created anew even if I deleted it beforehand) after every opendatahub-controller-manager-xxx pod restart.

**Comment by zdtsw** [comment]
i believe this has been changed and will be included in the ODH 2.5 release.
i just made a test:

- install the build from current incubation branch
- operator automatically created DSCI CR with default namespace `opendatahub` for both application and monitoring.
- edit CR set to `opendatahub-wen` for application namespace and `opendatahub-wen2 `for monitoring namespace.
- delete operator pod
- wait till pod running
- check DSCI CR again, got
```
spec:
  applicationsNamespace: opendatahub-wen1
  devFlags: {}
  monitoring:
    managementState: Managed
    namespace: opendatahub-wen2
```

**Comment by zdtsw** [comment]
i will have this issue closed for now, and the plan is to have ODH 2.5 release in coming days.
if you still encounter this bug, feel free to reopen the issue(or ping me in new comment) and i will take another look
